lib/safemode/core_jails.rb

```diff
@@ -94,56 +94,114 @@ def core_jail_class_methods(klass)
   # whitelisted methods for core classes ... kind of arbitrary selection
   @@methods_whitelist = {
   'Array' => %w(
+    all?
     any?
+    append
     assoc
     at
-    blank?
+    bsearch
+    bsearch_index
+    chunk
+    chunk_while
+    clear
     collect
     collect!
+    collect_concat
+    combination
     compact
     compact!
     concat
+    count
+    cycle
     delete
     delete_at
     delete_if
+    detect
+    dig
+    drop
+    drop_while
     each
+    each_cons
+    each_entry
     each_index
+    each_slice
+    each_with_index
+    each_with_object
     empty?
+    entries
     fetch
     fill
+    find
+    find_all
+    find_index
     first
+    flat_map
     flatten
     flatten!
-    hash
+    grep
+    grep_v
```

Comment:
Enables regexps essentially, I know that there is a TODO comment in the codebase like "are regexps safe"? Would be good idea to answer it here and also allow `=~`.